Source/Core/Core/HW/SystemTimers.cpp

```diff
@@ -102,6 +102,10 @@ static void DSPCallback(u64 userdata, s64 cyclesLate)
 static void AudioDMACallback(u64 userdata, s64 cyclesLate)
 {
 	int period = s_cpu_core_clock / (AudioInterface::GetAIDSampleRate() * 4 / 32);
+	if (SConfig::GetInstance().bEmulationSpeedFixAudio &&
```

Comment:
But it's expected to have perfect audio with this hack here and the disabled mixer, but this will break many games in a very unepexceted way. The expected ways are: audio-video desyncs in movies and no increase in gamespeed on several games.
IMO the better way would be to apply some kind of timestretching as we do within the OpenAL backend. It would still sounds worse than your proposed hack, but it won't break lots of games.